### PR TITLE
enable dnf-automatic, add packages for needs-restarting

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 
Enable dnf-automatic on Mariner, add packages for needs-restarting

###### Change Log  
- Enable dnf-automatic on Mariner, by turning on dnf-automatic-notifyonly and dnf-makecache timers.
- Build dnf-plugins-core (and its dependency dbus-python) to get needs-restarting command. This will be used to check if the system needs to be restarted due to recent version update.

###### Does this affect the toolchain?  
NO

###### Associated issues  
- None

###### Links to CVEs  
- None

###### Test Methodology
- Local build. Functionality tested on a Mariner machine on hyper-v using marketplace gen1 image.

### CODE REVIEW

File: SPECS/dbus-python/dbus-python.spec [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+%install
+%make_install
+
+%check
+pip3 install xml
+make check -k || (cat test-suite.log && false)
+
+%files -n python3-dbus
+%{_libdir}/python3.7/site-packages/*.so
```
Comment (hbeberman) [inline SPECS/dbus-python/dbus-python.spec:65]: Please add a %license line

File: SPECS/dbus-python/dbus-python.spec [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+%files devel
+%{_includedir}/dbus-1.0/dbus/dbus-python.h
+%{_libdir}/pkgconfig/dbus-python.pc
+%exclude %{_libdir}/python3.7/site-packages/*.la
+
+%changelog
+* Fri Jul 30 2021 Neha Agarwal <nehaagarwal@microsoft.com> - 1.2.16-6
+- Initial CBL-Mariner import from Fedora 34 (license: MIT)
+- Removed GPG signature, pkgconfig for BuildRequires
```
Comment (hbeberman) [inline SPECS/dbus-python/dbus-python.spec:76]: Double check that the dbus-python license is and add a License Verified statement in the changelog to confirm.

File: SPECS/dnf-plugins-core/dnf-plugins-core.spec [hunk 3]
```diff
@@ -0,0 +1,966 @@
...
+%{_mandir}/man1/yum-debug-restore.*
+%{_mandir}/man1/yum-groups-manager.*
+%{_mandir}/man1/yumdownloader.*
+%{_mandir}/man1/package-cleanup.*
+%{_mandir}/man1/dnf-utils.*
+%{_mandir}/man1/yum-utils.*
+
+%changelog
+* Fri Jul 30 2021 Neha Agarwal <nehaagarwal@microsoft.com> - 4.0.22-2
+- Initial CBL-Mariner import from rpm-software-management (license: GPLv2+)
```
Comment (hbeberman) [inline SPECS/dnf-plugins-core/dnf-plugins-core.spec:226]: Double check the license of the source code and add a License Verified.

File: SPECS/dnf/dnf.spec [hunk 4]
```diff
@@ -55,6 +56,7 @@ Systemd units that can periodically download package upgrades and apply them.
 
 %prep
 %setup -q
+sudo sed -i "s/emit_via = stdio/emit_via = motd/g" etc/dnf/automatic.conf
```
Comment (hbeberman) [inline SPECS/dnf/dnf.spec:59]: Is sudo required here? I thought the builder was root inside of the chroot.

**Review by hbeberman — CHANGES_REQUESTED** [review]

File: SPECS/dbus-python/dbus-python.spec (hunk 1 above)
Comment (neha170) [reply SPECS/dbus-python/dbus-python.spec:65]: Done

File: SPECS/dbus-python/dbus-python.spec (hunk 2 above)
Comment (neha170) [reply SPECS/dbus-python/dbus-python.spec:76]: Done

File: SPECS/dnf-plugins-core/dnf-plugins-core.spec (hunk 3 above)
Comment (neha170) [reply SPECS/dnf-plugins-core/dnf-plugins-core.spec:226]: Done

**Review by hbeberman — APPROVED** [review]

**Pull request merged**